client/web/reporter/src/views/Report/Builder.vue

```diff
@@ -1,383 +1,372 @@
-            :scenario.sync="currentScenario"
-          />
-        </template>
-      </configurator>
-    </b-modal>
-
-    <portal to="report-toolbar">
-      <editor-toolbar
-        :back-link="{ name: 'report.list' }"
-        :delete-disabled="!canDelete"
-        :save-disabled="!canUpdate"
-        :processing="processing"
-        :processing-save="processingSave"
-        :processing-delete="processingDelete"
-        @clone="handleReportCloning"
-        @delete="handleDelete"
-        @save="handleReportSave"
+        <selector
+          :items="datasources.types"
+          display-mode="text"
+          @select="addDatasource"
+        />
+      </b-modal>
+      <b-modal
+        v-model="scenarios.showConfigurator"
+        size="xl"
+        scrollable
+        :ok-title="$t('builder:scenarios.save')"
+        ok-variant="primary"
+        :title="$t('builder:scenarios.label')"
+        body-class="py-3"
+        no-fade
       >
-        <b-button
-          variant="light"
-          size="lg"
-          @click="createBlock"
+        <configurator
+          v-if="report"
+          :items="reportScenarios"
+          :current-index="scenarios.currentIndex"
+          draggable
+          @select="setCurrentScenario"
+          @add="addScenario()"
+          @delete="deleteCurrentScenario()"
         >
-          <font-awesome-icon
-            :icon="['fas', 'plus']"
-            size="sm"
-          />
-          {{ $t('general:label.add') }}
-        </b-button>
-      </editor-toolbar>
-    </portal>
+          <template #label="{ item: { label } }">
+            <span
+              class="d-inline-block text-truncate"
+            >
+              {{ label }}
+            </span>
+          </template>
+          <template #configurator>
+            <scenario-configurator
+              v-if="currentScenario"
+              :current-index="scenarios.currentIndex"
+              :datasources="reportDatasources"
+              :scenario.sync="currentScenario"
+            />
+          </template>
+        </configurator>
+      </b-modal>
+      <portal to="report-toolbar">
+        <editor-toolbar
+          :back-link="{ name: 'report.list' }"
+          :delete-disabled="!canDelete"
+          :save-disabled="!canUpdate"
+          :processing="processing"
+          :processing-save="processingSave"
+          :processing-delete="processingDelete"
+          :processing-clone="processingClone"
+          @clone="handleReportCloning"
+          @delete="handleDelete"
+          @save="handleReportSave"
+        >
+          <b-button
+            variant="light"
+            size="lg"
+            @click="createBlock"
+          >
+            <font-awesome-icon
+              :icon="['fas', 'plus']"
+              size="sm"
+            />
+            {{ $t('general:label.add') }}
+          </b-button>
+        </editor-toolbar>
+      </portal>
+    </div>
+
+    <div
```

Comment:
To prevent the toolbar and topbar flashing and/or users being stuck, move spinner before the grid.
Also make sure all buttons are disabled when processing except the back button. Ttopbar and toolbar; talking about reporter specific buttons like scenarios or datasources, not the ones that are always there,